src/components/Alert/Alert.tsx

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2023  New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import classNames from "classnames";
+import React, { PropsWithChildren } from "react";
+import { Icon } from "../Icon/Icon";
+import styles from "./Alert.module.css";
+
+type AlertProps<C extends React.ElementType> = {
+  as?: C;
+  type: "success" | "critical" | "info";
+  className?: string;
+  onClose?: (e: React.MouseEvent) => void;
+} & React.ComponentPropsWithoutRef<C>;
+
+export const Alert = <C extends React.ElementType = "div">({
+  as,
+  type,
+  children,
+  className,
+  onClose,
+  ...props
+}: PropsWithChildren<AlertProps<C>>): JSX.Element => {
+  const Component = as || "div";
```

Comment:
What do you see as the use case for this? Would it be good to lock it down to a subset of elements that make sense?

Reply:
You're probably right, reading through https://developer.mozilla.org/en-US/docs/Web/HTML/Element#text_content the two elements that make proper sense for this would be a `div` and ` p`.

But if the choices are that limited, wouldn't we be better off by only allowing paragraphs?

Reply:
I personally lean more towards locking things down, and only adding customisation like this when a real use-case presents itself.
There's bigger question of where we want this library to be - fully prescriptive <--> fully customisable. Not sure if that's been documented/discussed anywhere, also maybe a problem for future us? (Not trying to derail this PR!)